src/app/checklist/app-updates/app-updates.service.ts

```diff
@@ -0,0 +1,53 @@
+import { Injectable } from '@angular/core';
+import { MatSnackBar } from '@angular/material';
+import { SwUpdate } from '@angular/service-worker';
+import {
+  fromEvent as observableFromEvent,
+  interval,
+  merge as observableMerge,
+  Observable,
+  of as observableOf
+} from 'rxjs';
+import { map } from 'rxjs/operators';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class AppUpdatesService {
+  constructor(public snackbar: MatSnackBar, private serviceWorkerUpdates: SwUpdate) {}
+
+  checkUpdatesAuto() {
+    this.serviceWorkerUpdates.activated.subscribe(() => {
+      this.snackbar.open('App updated', 'OK', { duration: 2000 });
+    });
+
+    this.serviceWorkerUpdates.available.subscribe(() => {
+      const updateSnackbar = this.snackbar.open('Update available!', 'Update Now', {
+        duration: 2000
+      });
+
+      updateSnackbar.onAction().subscribe(() => {
+        updateSnackbar.dismiss();
+        window.location.reload();
+      });
+    });
+  }
+
+  checkUpdatesManually() {
+    this.snackbar.open(`You'll be notified if an update exists...`, 'OK', {
+      duration: 4000
+    });
+
+    interval(6 * 60 * 60).subscribe(() => {
```

Comment:
`6 * 60 * 60` should be refactored into a const. Something along the lines of `SW_UPDATE_INTERVAL`. This makes it very clear what this number does and what it's used for.